src/Domain/Syntax/Wildcard.php

```diff
@@ -22,15 +26,38 @@ public function __construct(
         $this->boost = $boost;
     }
 
+    public function setCaseInsensitive(bool $value): void
+    {
+        $this->caseInsensitive = $value;
+    }
+
+    public function setRewrite(string $value): void
+    {
+        $this->rewrite = $value;
+    }
+
     public function build(): array
     {
-        return [
-            'wildcard' => [
-                $this->field => [
-                    'value' => $this->value,
-                    'boost' => $this->boost,
-                ]
-            ]
+        $query = [
+            'value' => $this->value,
+            'boost' => $this->boost,
+            'case_insensitive' => $this->getCaseInsensitive(),
         ];
+
+        if (!empty($this->getRewrite())) {
+            $query['rewrite'] = $this->getRewrite();
+        }
+
+        return ['wildcard' => [ $this->field => $query ] ];
+    }
+
+    private function getCaseInsensitive(): bool
+    {
+        return $this->caseInsensitive;
+    }
+
+    private function getRewrite(): ?string
+    {
+        return $this->rewrite;
```

Comment:
It is totally fine to just use $this->rewrite in the other functions instead of these getters because the variables are private as well.